### PR TITLE
nixos/ollama: init

### PR DESCRIPTION
**Pull request by onny**
## Description of changes

Module for running [ollama](https://ollama.ai) via `ollama serve` in the background.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: nixos/modules/services/misc/ollama.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  config = lib.mkIf cfg.enable {
+
+    systemd = {
+      services.ollama = {
+        wantedBy = [ "multi-user.target" ];
+        description = "Server for local large language models";
+        after = [ "network.target" ];
+        environment = {
+          HOME = "/var/lib/ollama";
+          OLLAMA_MODELS = "/var/lib/ollama/models";
```
Comment (NickCao) [inline nixos/modules/services/misc/ollama.nix:24]: Please use systemd specifiers for these directories: https://www.freedesktop.org/software/systemd/man/latest/systemd.unit.html#Specifiers

File: nixos/modules/services/misc/ollama.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        };
+        serviceConfig = {
+          ExecStart = "${pkgs.ollama}/bin/ollama serve";
```
Comment (SuperSandro2000) [inline nixos/modules/services/misc/ollama.nix:27]:
```suggestion
          ExecStart = "${lib.getExe pkgs.ollama} serve";
```

**Comment by onny** [comment]
Thank you for reviewing :)

File: nixos/modules/services/misc/ollama.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+  cfg = config.services.ollama;
+
+in {
+
+  options = {
+    services.ollama = {
+      enable = lib.mkEnableOption (
+        lib.mdDoc "Server for local large language models"
+      );
```
Comment (natsukium) [inline nixos/modules/services/misc/ollama.nix:11]: I would like `package = mkPackageOption pkgs "ollama" { };`.

Comment (onny) [reply nixos/modules/services/misc/ollama.nix:11]: Thank you, added

**Review by SuperSandro2000 — APPROVED** [review]

**Pull request merged**